### PR TITLE
 [RHOAI] Create a CI build for ROCM Images for base, pytorch and tensorflow

### PR DESCRIPTION
**Pull request by dibryant**
Fixes for https://issues.redhat.com/browse/RHOAIENG-6371
Adding CI builds for AMD Base Pytorch and Tensorflow.

### CODE REVIEW

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml [hunk 1]
```diff
@@ -102,6 +102,26 @@ images:
...
+  to: rocm-ubi9-python-3.9
+- context_dir: jupyter/minimal/ubi9-python-3.9
+  dockerfile_path: Dockerfile
+  from: rocm-ubi9-python-3.9
+  to: rocm-jupyter-minimal-ubi9-python-3.9
+- context_dir: jupyter/data-science/ubi9-python-3.9
+  dockerfile_path: Dockerfile
+  from: rocm-jupyter-minimal-ubi9-python-3.9
+  to: rocm-jupyter-datascience-ubi9-python-3.9
+- context_dir: jupyter/amd/tensorflow/ubi9-python-3.9
```
Comment (atheo89) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml:117]:
```suggestion
- context_dir: jupyter/rocm/tensorflow/ubi9-python-3.9
```

**Comment by atheo89** [comment]
Thanks Diamond for opening this PR to change the AMD directory name to ROCM.
In order to be 100% functional these change keep in mind that you will need to update the dir name on the notebooks repo as well:
- https://github.com/opendatahub-io/notebooks/tree/main/amd
- https://github.com/opendatahub-io/notebooks/tree/main/jupyter/amd 

And also take a look on the dockerfile `label` to change any naming inconcistensy ex: https://github.com/opendatahub-io/notebooks/blob/main/jupyter/amd/pytorch/ubi9-python-3.9/Dockerfile#L5

**Comment by atheo89** [comment]
These changes are going to be introduced on downstream, could you refine a bit the title?
like: [RHOAI] Create a CI build for ROCM Images for base, pytorch and tensorflow

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml [hunk 2]
```diff
@@ -316,6 +344,39 @@ tests:
...
       RELEASE_VERSION: runtime-cuda-tensorflow-ubi9-python-3.9
     workflow: red-hat-data-services-ci-image-mirror
+- as: notebook-rocm-ubi9-python-3-9-image-mirror
+  postsubmit: true
+  run_if_changed: (base\/ubi9-python-3.9\/*)|(rocm\/ubi9-python-3.9\/*)
+  steps:
+    dependencies:
+      SOURCE_IMAGE_REF: rocm-ubi9-python-3.9
+    env:
+      IMAGE_REPO: workbench-images
```
Comment (harshad16) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml:354]:
```suggestion
      IMAGE_REPO: rocm-notebooks
```

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml [hunk 3]
```diff
@@ -316,6 +344,39 @@ tests:
...
+      RELEASE_VERSION: rocm-ubi9-python-3.9
+    workflow: opendatahub-io-ci-image-mirror
+- as: notebook-rocm-jupyter-tf-ubi9-python-3-9-image-mirror
+  postsubmit: true
+  run_if_changed: (rocm\/ubi9-python-3.9\/*)|(jupyter\/datascience\/ubi9-python-3.9\/*)|(jupyter\/rocm\/tensorflow\/ubi9-python-3.9\/*)
+  steps:
+    dependencies:
+      SOURCE_IMAGE_REF: rocm-jupyter-tensorflow-ubi9-python-3.9
+    env:
+      IMAGE_REPO: workbench-images
```
Comment (harshad16) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml:365]:
```suggestion
      IMAGE_REPO: rocm-notebooks
```

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml [hunk 4]
```diff
@@ -316,6 +344,39 @@ tests:
...
+      RELEASE_VERSION: rocm-jupyter-tensorflow-ubi9-python-3.9
+    workflow: opendatahub-io-ci-image-mirror
+- as: notebook-rocm-jupyter-pyt-ubi9-python-3-9-image-mirror
+  postsubmit: true
+  run_if_changed: (rocm\/ubi9-python-3.9\/*)|(jupyter\/datascience\/ubi9-python-3.9\/*)|(jupyter\/rocm\/pytorch\/ubi9-python-3.9\/*)
+  steps:
+    dependencies:
+      SOURCE_IMAGE_REF: rocm-jupyter-pytorch-ubi9-python-3.9
+    env:
+      IMAGE_REPO: workbench-images
```
Comment (harshad16) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml:376]:
```suggestion
      IMAGE_REPO: rocm-notebooks
```

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml [hunk 5]
```diff
@@ -346,6 +407,42 @@ tests:
       IMAGE_TAG: YearIndex
       RELEASE_VERSION: base-rhel9-python-3.9
     workflow: red-hat-data-services-ci-image-mirror
+- as: runtime-rocm-pytorch-ubi9-python-3-9-image-mirror
+  postsubmit: true
+  steps:
+    dependencies:
+      SOURCE_IMAGE_REF: runtime-rocm-pytorch-ubi9-python-3.9
+    env:
+      IMAGE_REPO: workbench-images
```
Comment (harshad16) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml:416]:
```suggestion
      IMAGE_REPO: rocm-notebooks
```

**Review by harshad16 — CHANGES_REQUESTED** [review]
Great work,
can we use `rocm-notebooks` for the IMAGE_REPO everywhere
and also can we push rocm-minimal, instead of only rocm

**Comment by jiridanek** [comment]
/pj-rehearse notebook-rocm-jupyter-tf-ubi9-python-3-9-image-mirror

**Comment by jiridanek** [comment]
aah, got it, the names of the jobs are like this! anyways, there's not too many of them created/touched, so running it all is perfectly fine

**Review by harshad16 — COMMENTED** [review]
All good, 
just missing some pr mirror and run-if-changed needs to be removed from postsubmit jobs.

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-release-2024a.yaml [hunk 6]
```diff
@@ -316,6 +344,36 @@ tests:
...
     workflow: red-hat-data-services-ci-image-mirror
+- as: notebook-rocm-jupyter-minimal-ubi9-python-3-9-image-mirror
+  postsubmit: true
+  steps:
+    dependencies:
+      SOURCE_IMAGE_REF: rocm-jupyter-minimal-ubi9-python-3.9
+    env:
+      IMAGE_REPO: rocm-notebooks
+      IMAGE_TAG: YearIndex
+      RELEASE_VERSION: rocm-jupyter-minimal-ubi9-python-3.9
```
Comment (harshad16) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-release-2024a.yaml:355]:
```suggestion
      RELEASE_VERSION: rocm-jupyter-minimal-ubi9-python-3.9-2024a
```

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-release-2024a.yaml [hunk 7]
```diff
@@ -316,6 +344,36 @@ tests:
...
+    workflow: opendatahub-io-ci-image-mirror
+- as: notebook-rocm-jupyter-tf-ubi9-python-3-9-image-mirror
+  postsubmit: true
+  steps:
+    dependencies:
+      SOURCE_IMAGE_REF: rocm-jupyter-tensorflow-ubi9-python-3.9
+    env:
+      IMAGE_REPO: rocm-notebooks
+      IMAGE_TAG: YearIndex
+      RELEASE_VERSION: rocm-jupyter-tensorflow-ubi9-python-3.9
```
Comment (harshad16) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-release-2024a.yaml:365]:
```suggestion
      RELEASE_VERSION: rocm-jupyter-tensorflow-ubi9-python-3.9-2024a
```

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-release-2024a.yaml [hunk 8]
```diff
@@ -316,6 +344,36 @@ tests:
...
+    workflow: opendatahub-io-ci-image-mirror
+- as: notebook-rocm-jupyter-pyt-ubi9-python-3-9-image-mirror
+  postsubmit: true
+  steps:
+    dependencies:
+      SOURCE_IMAGE_REF: rocm-jupyter-pytorch-ubi9-python-3.9
+    env:
+      IMAGE_REPO: rocm-notebooks
+      IMAGE_TAG: YearIndex
+      RELEASE_VERSION: rocm-jupyter-pytorch-ubi9-python-3.9
```
Comment (harshad16) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-release-2024a.yaml:375]:
```suggestion
      RELEASE_VERSION: rocm-jupyter-pytorch-ubi9-python-3.9-2024a
```

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-release-2024a.yaml [hunk 9]
```diff
@@ -346,6 +404,24 @@ tests:
...
       RELEASE_VERSION: base-rhel9-python-3.9-2024a
     workflow: red-hat-data-services-ci-image-mirror
+- as: runtime-rocm-pytorch-ubi9-python-3-9-image-mirror
+  postsubmit: true
+  steps:
+    dependencies:
+      SOURCE_IMAGE_REF: runtime-rocm-pytorch-ubi9-python-3.9
+    env:
+      IMAGE_REPO: rocm-notebooks
+      RELEASE_VERSION: runtime-rocm-pytorch-ubi9-python-3.9
```
Comment (harshad16) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-release-2024a.yaml:414]:
```suggestion
      RELEASE_VERSION: runtime-rocm-pytorch-ubi9-python-3.9-2024a
```

**Review by harshad16 — CHANGES_REQUESTED** [review]
some additional changes needed

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml [hunk 10]
```diff
@@ -102,6 +102,26 @@ images:
...
   to: runtime-cuda-tensorflow-ubi9-python-3.9
+- context_dir: rocm/ubi9-python-3.9
+  dockerfile_path: Dockerfile
+  from: base-ubi9-python-3.9
+  to: rocm-ubi9-python-3.9
+- context_dir: jupyter/minimal/ubi9-python-3.9
+  dockerfile_path: Dockerfile
+  from: rocm-ubi9-python-3.9
+  to: rocm-jupyter-minimal-ubi9-python-3.9
+- context_dir: jupyter/data-science/ubi9-python-3.9
```
Comment (harshad16) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml:113]:
```suggestion
- context_dir: jupyter/datascience/ubi9-python-3.9
```

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-release-2024a.yaml [hunk 11]
```diff
@@ -102,6 +102,26 @@ images:
...
   to: runtime-cuda-tensorflow-ubi9-python-3.9
+- context_dir: rocm/ubi9-python-3.9
+  dockerfile_path: Dockerfile
+  from: base-ubi9-python-3.9
+  to: rocm-ubi9-python-3.9
+- context_dir: jupyter/minimal/ubi9-python-3.9
+  dockerfile_path: Dockerfile
+  from: rocm-ubi9-python-3.9
+  to: rocm-jupyter-minimal-ubi9-python-3.9
+- context_dir: jupyter/data-science/ubi9-python-3.9
```
Comment (harshad16) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-release-2024a.yaml:113]:
```suggestion
- context_dir: jupyter/datascience/ubi9-python-3.9
```

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml [hunk 12]
```diff
@@ -102,6 +102,26 @@ images:
...
+  dockerfile_path: Dockerfile
+  from: rocm-jupyter-minimal-ubi9-python-3.9
+  to: rocm-jupyter-datascience-ubi9-python-3.9
+- context_dir: jupyter/rocm/tensorflow/ubi9-python-3.9
+  dockerfile_path: Dockerfile
+  from: jupyter-datascience-ubi9-python-3.9
```
Comment (harshad16) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml:119]:
```suggestion
  from: rocm-jupyter-datascience-ubi9-python-3.9
```

File: ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml [hunk 13]
```diff
@@ -102,6 +102,26 @@ images:
...
+  to: rocm-jupyter-tensorflow-ubi9-python-3.9
+- context_dir: jupyter/rocm/pytorch/ubi9-python-3.9
+  dockerfile_path: Dockerfile
+  from: jupyter-datascience-ubi9-python-3.9
```
Comment (harshad16) [inline ci-operator/config/red-hat-data-services/notebooks/red-hat-data-services-notebooks-main.yaml:123]:
```suggestion
  from: rocm-jupyter-datascience-ubi9-python-3.9
```

**Review by harshad16 — CHANGES_REQUESTED** [review]
final change, this are required.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

thanks :100:
